server/src/main/java/org/opensearch/cluster/metadata/DecommissionAttributeMetadata.java

```diff
@@ -0,0 +1,253 @@
+    public DecommissionAttributeMetadata(DecommissionAttribute decommissionAttribute) {
+        this.decommissionAttribute = decommissionAttribute;
+        this.status = DecommissionStatus.INIT;
+    }
+
+    /**
+     * Returns the current decommissioned attribute
+     *
+     * @return decommissioned attributes
+     */
+    public DecommissionAttribute decommissionAttribute() {
+        return this.decommissionAttribute;
+    }
+
+    /**
+     * Returns the current status of the attribute decommission
+     *
+     * @return attribute type
+     */
+    public DecommissionStatus status() {
+        return this.status;
+    }
+
+    public DecommissionAttributeMetadata withUpdatedStatus(
+        DecommissionAttributeMetadata metadata,
+        DecommissionStatus status) {
+        return new DecommissionAttributeMetadata(
+            metadata.decommissionAttribute(),
+            status
+        );
+    }
+
+    /**
+     * Creates a new instance with a updated attribute value.
+     *
+     * @param metadata       current metadata
+     * @param attributeValue new attribute value
+     * @return new instance with updated attribute value and status as DecommissionStatus.INIT
+     */
+    public DecommissionAttributeMetadata withUpdatedAttributeValue(
+        DecommissionAttributeMetadata metadata,
+        String attributeValue
+    ) {
+        return new DecommissionAttributeMetadata(
+            new DecommissionAttribute(metadata.decommissionAttribute, attributeValue),
+            DecommissionStatus.INIT
+        );
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        DecommissionAttributeMetadata that = (DecommissionAttributeMetadata) o;
+
+        if (!status.equals(that.status)) return false;
+        return decommissionAttribute.equals(that.decommissionAttribute);
+    }
+
+    /**
+     * Checks if this instance and the given instance share the same decommissioned attributeName
+     * and only differ in the attributeValue {@link DecommissionAttribute#attributeValue()}
+     *
+     * @param other other decommission attribute metadata
+     * @return {@code true} iff both instances contain the same attributeName
+     */
+    public boolean equalsIgnoreValue(@Nullable DecommissionAttributeMetadata other) {
+        if (other == null) {
+            return false;
+        }
+        if (!status.equals(other.status)) return false;
+        return decommissionAttribute.equalsIgnoreValues(other.decommissionAttribute);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(attributeType, decommissionAttribute, status);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getWriteableName() {
+        return TYPE;
+    }
+
+    @Override
+    public Version getMinimalSupportedVersion() {
+        return Version.CURRENT.minimumCompatibilityVersion();
```

Comment:
is it better to put a specific version, other wise in future when the Current changes, this would need to be changed as well.

Reply:
Yes. Making it V_2_3_0 in next commit.